<a href="https://colab.research.google.com/github/jendives2000/probabilities-statistics/blob/main/Hypothesis%20Proportion%20Test%20Practice%201.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

You work as an analyst for a retail company that is interested in understanding customer satisfaction. The company claims that 80% of its customers are satisfied with their service. However, after a recent change in their return policy, you suspect that the satisfaction rate might have decreased. You decide to test this claim.


---



**Given:**

*   Population Proportion ($p_0$): 0.80 (80%)
  *   Successes = S = $p_0$

*   Sample Size (𝑛): 150 customers

*   Sample Proportion ($\hat{p}$): You will calculate it based on the provided sample data.

*   Significance Level (𝛼): 0.05



---



**Hypotheses**:
*   Null Hypothesis ($H_0$): 𝑝≥0.80

  *   (At least 80% of customers are satisfied)

*   Alternative Hypothesis ($H_a$): 𝑝<0.80
  *   (The proportion of satisfied customers is less than 80%)


---



We were not given the sample proportion (p hat). Let's calculate it:

$\hat{p} = \frac{S}{n}$

In [8]:
successes = 80
s_float = successes / 100

n = 150
p_hat = round(successes / n, 3)
print(f'\nThe sample proportion (p hat) is:\n\t{round(p_hat, 3)}')


The sample proportion (p hat) is:
	0.533




---


Now we know that:

$\hat{p}$ = 0.533

Let's find sigma sub p hat:

$\sigma_{\hat{p}}$ = $\sqrt{\frac{\hat{p}(1-\hat{p})}{n}}$


In [6]:
import math

sigmasub_phat = math.sqrt((p_hat * (1 - p_hat)) / n)
print(f'\nThe sample standard deviation (sigma sub p hat) is:\n\t{round(sigmasub_phat, 3)}')


The sample standard deviation (sigma sub p hat) is:
	0.041




---


We can now calculate the z score and then get the p value from it.

In [11]:
from scipy.stats import norm

z_score = (p_hat - s_float) / sigmasub_phat
print(f'\nThe z score is:\n\t{round(z_score, 3)}')
p_value = norm.cdf(z_score)

print(f'\nWith this z score, our p value is:\n\t{round(p_value, 6)}')


The z score is:
	-6.554

With this z score, our p value is:
	0.0




---


Now we have our p value, we need to check this:

p < α

And so we have:

0.00 < 0.05

This is **true**.

# CONCLUSIONS: 

In conclusion, we can **reject the null hypothesis** in favor of the alternative hypothesis.

This means we did not have enough evidence to say that the proportion of satisfied customer is equal to or more than 80%.
It is **more likely that the proportion of satisfied customers is strictly less than 80%.**

## Recommendations:  

### BEFORE THE POLICIY CHANGE:
*   Conduct A/B Testing on Policy Changes:  
Before rolling out any further policy changes company-wide, conduct A/B testing on smaller customer groups to understand the impact more clearly.

     - Action Plan:  
     Create two groups with slightly different return policies or incentives and monitor their satisfaction levels, return rates, and overall sales to determine which approach yields better results.  
     
*   Explore Predictive Analytics:  
Utilize predictive analytics models to anticipate customer behavior, such as predicting which customers are most likely to be dissatisfied or to return products.

     - Action Plan:  
     Work with the data science team to build models that help in early identification of at-risk customers and proactively address their concerns to improve retention.
### AFTER:
*   a. Segment of Clients who returned products Vs those who did not:  
A more targeted approach will further refine our analysis. By comparing the proportion of satisfied clients who returned a product to the proportion of those who did not, we can further assess the impact of the change in the return policy.  
I suggest that the marketing team starts a feedback campaign on 100 clients from the 1st segment and the data team conducts a sentiment analysis on it to helps us identify pain points. This helps the sales team to better prevent clients' loss. 

*   b. Analyze Customer Demographics:
Segment the customer base by demographic factors such as age, gender, and location. This will help identify specific groups that are more likely to be dissatisfied, allowing for more personalized interventions. For instance, younger customers might value different aspects of the product or service compared to older customers.

*   c. Monitor Satisfaction Over Time:
Implement a longitudinal study to track changes in customer satisfaction over time. By continuously monitoring satisfaction levels before and after any policy changes, you can better assess the effectiveness of these changes and quickly identify trends or emerging issues.

*   d. Enhance Customer Support:
Strengthen the customer support system by offering more personalized assistance, especially to those who have returned products. Ensuring that customer complaints and concerns are addressed promptly and effectively could help increase satisfaction and reduce product returns.

*   e. Improve Product Quality and Communication:
Assess and improve the quality of products, especially those with higher return rates. Additionally, review and enhance communication strategies to ensure customers have clear and accurate expectations of the products they purchase.


---

